adding code limit question to FAQ (#69)
risc0/website: docs/faq.md

-------------------------------
## Contents
- [FAQ](#faq)
  - [Contents](#contents)
  - [ZKP System](#zkp-system)
  - [Code Project Help](#code-project-help)

>   - [<br/>](#)

Reviewer: At least to my eye, bullets of `<br/>`s render pretty awkwardly

</details>
<br/>
<details closed>
<summary>
Q:
I've encountered an error while running my project for the RISC Zero zkVM! It looks like this:

* Error A:
<pre>
<code>thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Exception &#123; what: "Expand failed: at max steps" &#125;', starter/src/main.rs:7:32</code>
</pre>
* Error B:
<pre>
<code>thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Exception &#123; what: "circuit.validCode(codeMerkle.getRoot())" &#125;', starter/src/main.rs:7:32
</code>
</pre>

What is happening and how can I fix it?
</summary>
<br/>

> A: Both errors arise when the zkVM runs out of available instruction cycles. Error A arises if your guest zkVM program took more instruction cycles to run than were available. Error B comes about if you've only used <i>slightly</i> more instruction cycles than the zkVM can process; the zkVM has completed program execution, but it ran out of steam during receipt validation.

Reviewer: Maybe replace the metaphor/idiom with a literal description of the failure? 
```suggestion
A: Both errors arise when the zkVM runs out of available instruction cycles. Error A arises if your guest zkVM program took more instruction cycles to run than were available. Error B comes about if you've only used <i>slightly</i> more instruction cycles than the zkVM can process; the zkVM has completed program execution, but it ran out of cycles during receipt validation.
```

<br/><br/>

> In both cases, you can manually increase the number of available instruction cycles by changing the `code_limit` build option. To do this, edit <a href ="https://github.com/risc0/risc0-rust-starter/blob/main/methods/build.rs">methods/build.rs</a> to call <code>embed_methods_with_options</code> rather than <code>embed_methods</code>. We've opted for a <code>no-std</code> build in the snippet below, so use <code>std: true</code> if your guest program relies on standard:

Reviewer: Is it better to give a somewhat more general statement of what to replace (as in this suggestion)?
```suggestion
In both cases, you can manually increase the number of available instruction cycles by changing the `code_limit` build option. To do this replace <code>embed_methods</code> with a call to <code>embed_methods_with_options</code> (In our example code, this call happens in <a href ="https://github.com/risc0/risc0-rust-starter/blob/main/methods/build.rs">methods/build.rs</a>). We've opted for a <code>no-std</code> build in the snippet below, so use <code>std: true</code> if your guest program relies on standard:
```

<br/><br/>
<code>

> // The default code_limit is 12. Increase this value up to a maximum of 16.

Reviewer 2: Can we add a note translating this number into a more dev-friendly metric? How should I think about this limit as a developer?

Reviewer 2: Not a blocking suggestion -- this may be more complicated than I realize

> let options_map = HashMap::from([("methods-guest", risc0_build::GuestOptions{ code_limit: 16, features: vec![], std: false })]);

Reviewer: The structure of `risc0_build::GuestOptions` varies based on whether you are on 0.11 or the current `HEAD` of `main`. What you wrote here is is for `HEAD`, and will not work for 0.11.

Author: Updated to match v0.11 and tested
